server/index.js

```diff
@@ -1,21 +1,119 @@
 const { PrismaClient } = require('@prisma/client')
+const { Client } = require('pg')
 const prisma = new PrismaClient()
+const WebSocket = require('ws')
+const pgClient = new Client({
+    connectionString: "postgresql://postgres:Vanessa97723$@localhost:5432/hobbydata?schema=public"
+})
 require('dotenv').config();
 var cors = require('cors')
 const express = require('express')
 const bcrypt = require('bcrypt');
 const saltRounds = 14;
 const app = express()
 const PORT = 3000
+const wss = new WebSocket.Server({ port: 8080 }, () => {
+    console.log("Websocket server is running on ws://localhost:8080")
+})
 
 app.use(express.json());
 app.use(cors());
 
+const clients = {}
+
+// NOTIFICATONS
+
+pgClient.connect()
+
+pgClient.query('LISTEN new_user')
+pgClient.query('LISTEN new_post')
+
+pgClient.on('notification', async (msg) => {
+    console.log('received notifictiion:')
+    const payload = JSON.parse(msg.payload)
+    const data = {
+        type: msg.channel,
+        ...payload,
+    }
+    if (msg.channel === 'new_user') {
+        const { hobbyId, username } = payload;
```

Comment:
One idea to make this code cleaner is to do something like:

if (msg.channel === 'new_user')
   handleNewUserNotification(hobbyId, username)
else if (msg.channel === 'new_post')
    handleMsgChannelNotification(hobbyId, caption, username)